### ISSUE TITLE
Be more lenient when parsing for "---" to allow for extra "-"  

### ISSUE THREAD

**Issue by sangee2004** [gptscript-ai/gptscript#23]
Currently we get the following error message when we have extra "-" in the tools delimiter "---"
```
2024/02/12 11:35:36 failed resolving <tool name> at  <gptscrpit name>: can not load tools path=. name=<tool name>
```

It would be good to be good to more leninet when parsing for "---" to allow for extra "-"  .

**Comment by ibuildthecloud** [comment]
This should be fixed in 0.1

**Comment by sangee2004** [comment]
Tested with gptscript version `v0.1.0-bbf61d32-dirty`
We are now allowed to have more number of "-" as delimiters. There needs to be a minimum of 3 "-" (---) to  be a valid delimiter but any string with more > 3 "-" like `-----` is allowed delimiter.
